### PR TITLE
terraform project support

### PR DESCRIPTION
**Pull request by mk-armah**
# Description

What - Adding support for Terraform Projects

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: integrations/terraform-cloud/.port/resources/blueprints.json [hunk 1]
```diff
@@ -123,9 +123,42 @@
...
+            "type": "string",
+            "title": "Project Name",
+            "description": "The name of the Terraform Cloud project"
+          },
+          "permissions": {
+            "type": "object",
+            "title": "Permissions",
+            "description": "The permisssions on the project"
+          },
+          "organizationId": {
```
Comment (Tankilevitch) [inline integrations/terraform-cloud/.port/resources/blueprints.json:150]: change the order to be in the order of relations for cleaner reading of the blueprints json.

project > workspace > cloudStateversion > cloudRun

File: integrations/terraform-cloud/.port/resources/blueprints.json [hunk 2]
```diff
@@ -123,9 +123,42 @@
...
+            "type": "string",
+            "title": "Organization ID",
+            "description": "The ID of the organization the project belongs to"
+          }
+        }
+      },
+      "relations":{},
+      "mirrorProperties":{},
```
Comment (Tankilevitch) [inline integrations/terraform-cloud/.port/resources/blueprints.json:158]: Relation between project and cluster?

File: integrations/terraform-cloud/.port/resources/blueprints.json [hunk 3]
```diff
@@ -123,9 +123,42 @@
...
+            "description": "The permisssions on the project"
+          },
+          "organizationId": {
+            "type": "string",
+            "title": "Organization ID",
+            "description": "The ID of the organization the project belongs to"
+          }
+        }
+      },
+      "relations":{},
```
Comment (Tankilevitch) [inline integrations/terraform-cloud/.port/resources/blueprints.json:157]: keep the convention of relations being last property

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
minor

File: integrations/terraform-cloud/.port/resources/blueprints.json [hunk 4]
```diff
@@ -123,9 +123,42 @@
             "target":"terraformCloudStateVersion",
             "required":false,
             "many":false
-         }
+         },
+         "project":{"title":"Terraform Cloud Project",
```
Comment (yairsimantov20) [inline integrations/terraform-cloud/.port/resources/blueprints.json:127]: Format your json please

File: integrations/terraform-cloud/CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,17 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+
+### Features
+
+- Added Support for Terraform Project (0.1.3)
+
+### Improvements
+
+- Fixed bug failing Terraform Workspaces sync (0.1.3)
+- Rate Limit Exception Handling (0.1.3)
+
```
Comment (yairsimantov20) [inline integrations/terraform-cloud/CHANGELOG.md:19]: Be more verbose please

File: integrations/terraform-cloud/main.py [hunk 6]
```diff
@@ -53,6 +54,14 @@ async def enrich_state_versions_with_output_data(
         return enriched_state_versions
 
 
+@ocean.on_resync(ObjectKind.PROJECT)
+async def resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    terraform_client = init_terraform_client()
+    async for project in terraform_client.get_paginated_projects():
+        logger.info(f"Received {len(project)} batch {kind}s")
+        yield project
```
Comment (yairsimantov20) [inline integrations/terraform-cloud/main.py:62]: Better naming please
This is not a single project

File: integrations/terraform-cloud/pyproject.toml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "terraform-cloud"
-version = "0.1.1"
```
Comment (yairsimantov20) [inline integrations/terraform-cloud/pyproject.toml:3]: ?
Why 2 versions?

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]
Comments

File: integrations/terraform-cloud/pyproject.toml (hunk 7 above)
Comment (mk-armah) [reply integrations/terraform-cloud/pyproject.toml:3]: @yairsimantov20 , sorry i don't understand, i'm seeing only one version. 0.1.3

**Pull request closed without merging**